zebra-state/src/service.rs

```diff
@@ -720,28 +720,31 @@ impl StateService {
             return rsp_rx;
         }
 
-        if self.finalized_block_write_sender.is_none() {
-            // Wait until block commit task is ready to write non-finalized blocks before dequeuing them
-            self.send_ready_non_finalized_queued(parent_hash);
+        self.send_ready_non_finalized_queued(parent_hash);
 
-            let finalized_tip_height = self.read_service.db.finalized_tip_height().expect(
+        rsp_rx
+    }
+
+    /// Returns `true` if `hash` is a valid previous block hash for new non-finalized blocks.
+    fn can_fork_chain_at(&mut self, hash: &block::Hash) -> bool {
```

Comment:
This PR changes syncing behaviour, can you run a full sync test before we merge it?
You'll have to do it manually, or get Gustavo to fix #6568 first.

Reply:
I fixed #6568, I hope!

Reply:
Since there's a sync risk, this PR should go at the end of the PR backlog.

Reply:
Ok great that seems fine.

The scheduled full syncs usually complete in just over 2 days.